File: client/cmd/token/account.go
package token

import (
	"fmt"

	"github.com/iden3/go-iden3-crypto/poseidon"
	"github.com/spf13/cobra"
	"source.quilibrium.com/quilibrium/monorepo/client/utils"
)

var AccountCmd = &cobra.Command{
	Use:   "account",
	Short: "Shows the account address of the managing account",
	Run: func(cmd *cobra.Command, args []string) {
		peerId := utils.GetPeerIDFromConfig(NodeConfig)
		addr, err := poseidon.HashBytes([]byte(peerId))
		if err != nil {
			panic(err)
		}

		addrBytes := addr.FillBytes(make([]byte, 32))
		fmt.Printf("Account: 0x%x\n", addrBytes)
	},
}

File: crates/quil-client/src/commands/token/account.rs
//! `qclient token account` — print the managing account address.
//!
//! Port of `client/cmd/token/account.go`: `poseidon(peerId)` → 32 bytes,
//! printed as `0x…`.

use super::TokenCtx;

pub fn run(tc: &TokenCtx) -> anyhow::Result<()> {
    let addr = quil_crypto::poseidon::hash_bytes_to_32(&tc.peer_id_bytes)
        .map_err(|e| anyhow::anyhow!("poseidon account address: {e}"))?;
    println!("Account: 0x{}", hex::encode(addr));
    Ok(())
}
